src/brs/BurstGUI.java

```diff
@@ -0,0 +1,308 @@
+     */
+    public static void addToClasspath(String s) {
+        try {
+            File f = new File(s);
+            URI u = f.toURI();
+            URLClassLoader urlClassLoader = (URLClassLoader) ClassLoader.getSystemClassLoader();
+            Class<URLClassLoader> urlClass = URLClassLoader.class;
+            Method method = urlClass.getDeclaredMethod("addURL", URL.class);
+            method.setAccessible(true);
+            method.invoke(urlClassLoader, u.toURL());
+        } catch (Exception e) {
+            LOGGER.error("Could not add path \"" + s + "\" to classpath", e);
+        }
+    }
+
+    private static void shutdown() {
+        userClosed = true;
+        if (trayIcon != null && SystemTray.isSupported()) {
+            SystemTray.getSystemTray().remove(trayIcon);
+        }
+        System.exit(0); // BRS shutdown handled by exit hook
+    }
+
+    private static void showTrayIcon() {
+        if (trayIcon == null) { // Don't start running in tray twice
+            trayIcon = createTrayIcon();
+            if (trayIcon != null) {
+                stage.setOnCloseRequest(event -> hideWindow());
+            } else {
+                stage.show();
+                stage.setOnCloseRequest(event -> shutdown());
+            }
+        }
+    }
+
+    /**
+     * @return A tray icon if successful or null otherwise
+     */
+    private static TrayIcon createTrayIcon() {
+        try {
+            SystemTray systemTray = SystemTray.getSystemTray();
+            PopupMenu popupMenu = new PopupMenu();
+
+            MenuItem openWebUiButton = new MenuItem("Open Web GUI");
+            MenuItem showItem = new MenuItem("Show BRS output");
+            MenuItem shutdownItem = new MenuItem("Shutdown BRS");
+
+            openWebUiButton.addActionListener(e -> openWebUi());
+            showItem.addActionListener(e -> showWindow());
+            shutdownItem.addActionListener(e -> shutdown());
+
+            popupMenu.add(openWebUiButton);
+            popupMenu.add(showItem);
+            popupMenu.add(shutdownItem);
+
+            TrayIcon newTrayIcon = new TrayIcon(ImageIO.read(BurstGUI.class.getResourceAsStream("/images/burst_overlay_logo.png")), "Burst Reference Software", popupMenu);
+            newTrayIcon.setImage(newTrayIcon.getImage().getScaledInstance(newTrayIcon.getSize().width, -1, Image.SCALE_SMOOTH));
+            newTrayIcon.addActionListener(e -> openWebUi());
+            systemTray.add(newTrayIcon);
+            return newTrayIcon;
+        } catch (Exception e) {
+            LOGGER.error("Could not create tray icon", e);
+            return null;
+        }
+    }
+
+    private static void showWindow() {
+        Platform.runLater(stage::show);
+    }
+
+    private static void hideWindow() {
+        Platform.runLater(stage::hide);
+    }
+
+    private static void openWebUi() {
+        try {
+            Desktop.getDesktop().browse(new URI("http://localhost:8125"));
```

Comment:
Should this use config values in case user is running on a different port?

Reply:
Good idea!